airflow/jobs.py

```diff
@@ -228,7 +228,9 @@ def reset_state_for_orphaned_tasks(self, dag_run, session=None):
         tis.extend(dag_run.get_task_instances(state=State.QUEUED, session=session))
 
         for ti in tis:
-            if ti.key not in queued_tis and ti.key not in running:
+            # check if '.' is in dag_run's dag_id because scheduler should not reset the state
+            # of subdag task
+            if ti.key not in queued_tis and ti.key not in running and '.' not in ti.dag_id:
```

Comment:
Let's check if the DAG is a subdag directly (dag.is_subdag) instead of checking the presence of a period in the dag_id.